package/src/components/FulfillmentOptionsCheckoutAction/v1/FulfillmentOptionsCheckoutAction.js

```diff
@@ -128,9 +128,13 @@ class FulfillmentOptionsCheckoutAction extends Component {
       return selectedFulfillmentOption.fulfillmentMethod._id;
     }
 
+    // Apollo Client returns immutable Objects, therefore it's necessary to
+    // clone the fulfillment options before sorting.
+    const clonedFulfillmentOptions = availableFulfillmentOptions.slice();
+
     // If a selection has not been made yet, default to cheapest
     if (availableFulfillmentOptions && availableFulfillmentOptions.length > 0) {
-      return availableFulfillmentOptions.sort((itemA, itemB) => itemA.price.amount - itemB.price.amount)[0].fulfillmentMethod._id;
+      return clonedFulfillmentOptions.sort((itemA, itemB) => itemA.price.amount - itemB.price.amount)[0].fulfillmentMethod._id;
```

Comment:
@willopez I was just thinking... Cloning + sorting actually might be kind of expensive. I don't know how often this reruns, but it is usually better to be safe than sorry.

Since we don't actually need a sorted array -- we just need the cheapest one -- you could use a simple reduce instead.

```js
const cheapestOption = availableFulfillmentOptions.reduce((cheapest, option) => {
  if (!cheapest || option.price.amount < cheapest.price.amount) {
    return option;
  }
  return cheapest;
}, null);
```

That way you do not need to clone it either.